ocaml/idl/dm_api.ml

```diff
@@ -200,12 +202,15 @@ let check api emergency_calls =
   let (_ : obj list) =
     map_field
       (function
```

Comment:
The double function is a bit convoluted and add indentation, might be worth using to be less disruptive

```ocaml
      (fun _ -> function
        | {ty= Ref _; field_has_effect= true} ->
            failwith
              "Can't have a Ref field with a side-effect: it makes the \
               destructors too complicated"
        | x ->
            x
```